lib/user.rb

```diff
@@ -0,0 +1,42 @@
+class User 
+
+  attr_accessor :name, :job, :email, :address, :pin, :logged_in
+
+  def initialize(name, job, email, address, pin)
+    @name = name
+    @job = job
+    @email = email
+    @address = address
+    set_pin(pin)
+    @logged_in = false
+  end
+
+  def set_pin(pin)
+    if valid_pin?(pin)
+      @pin = pin
+      puts "PIN set successfully. [#{pin}]"
+      true
+    else
+      puts "Invalid PIN. It must be a 4-digit number."
+      false
+    end
```

Comment:
I would suggest checking for errors first and returning from the method when finding one. This way we can avoid if-else clauses:
```suggestion
    unless valid_pin?(pin)
      puts "Invalid PIN. It must be a 4-digit number."
      return false
    end
    
    @pin = pin
    puts "PIN set successfully. [#{pin}]"
    true
```